testkit/src/main/scala/org/bitcoins/testkit/core/gen/ScriptGenerators.scala

```diff
@@ -219,7 +220,8 @@ sealed abstract class ScriptGenerators extends BitcoinSLogger {
 
   def p2shScriptPubKey: Gen[(P2SHScriptPubKey, Seq[ECPrivateKey])] =
     for {
-      (randomScriptPubKey, privKeys) <- randomNonP2SHScriptPubKey
+      (randomScriptPubKey, privKeys) <- randomNonP2SHScriptPubKey.suchThat(
+        _._1.compactSizeUInt.toLong + CompactSizeUInt(UInt64(520)).bytes.length < 520L)
```

Comment:
this is really used in a lot of different places at this point, do we need it everywhere? At minimum we should isolate this generator check into a single helper method.